xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -170,46 +182,140 @@ public void handleNameResolutionError(Status error) {}
 
             @Override
             public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-              if (xdsClient == null) {
-                ManagedChannel channel = initLbChannel(helper, newBalancerName, channelCredsList);
-                xdsClient  = new XdsComms2(
-                    channel, helper, new ExponentialBackoffPolicy.Provider(),
-                    GrpcUtil.STOPWATCH_SUPPLIER, node);
-                localityStore = localityStoreFactory.newLocalityStore(helper, lbRegistry);
+              if (xdsClientRef != null) {
```

Comment:
You don't need this checks.

1. This half-baked case will never happen.
2. Do not rely on checks to help you reveal bugs at runtime. Checks should only be used for things affected by external calls/responses and things out of your implementation's control.

Reply:
Removed checks.